x-pack/solutions/search/plugins/workchat/wci_salesforce/server/integration/mcp_server.ts

```diff
@@ -5,69 +5,200 @@
+      owner: z.string().optional().describe('Email of the case owner/assignee to filter results'),
+      priority: z
+        .enum(priorityValues.length ? (priorityValues as [string, ...string[]]) : [''])
+        .optional()
+        .describe(
+          `Case priority level${
+            priorityValues.length ? ` (one of: ${priorityValues.join(', ')})` : ''
+          }`
+        ),
+      status: z
+        .enum(statusValues.length ? (statusValues as [string, ...string[]]) : [''])
+        .optional()
+        .describe(
+          `Current status of the case${
+            statusValues.length ? ` (one of: ${statusValues.join(', ')})` : ''
+          }`
+        ),
+      closed: z.boolean().optional().describe('Filter by case closure status (true/false)'),
+      createdAfter: z
+        .string()
+        .optional()
+        .describe('Return cases created after this date (format: YYYY-MM-DD)'),
+      createdBefore: z
+        .string()
+        .optional()
+        .describe('Return cases created before this date (format: YYYY-MM-DD)'),
+      semanticQuery: z
+        .string()
+        .optional()
+        .describe('Natural language query to search case content semantically'),
+      updatedAfter: z
+        .string()
+        .optional()
+        .describe('Return cases updated after this date (format: YYYY-MM-DD)'),
+      updatedBefore: z
+        .string()
+        .optional()
+        .describe('Return cases updated before this date (format: YYYY-MM-DD)'),
+    },
+    async ({
+      id,
+      size = 10,
+      owner,
+      priority,
+      closed,
+      caseNumber,
+      createdAfter,
+      createdBefore,
+      semanticQuery,
+      status,
+      updatedAfter,
+      updatedBefore,
+    }) => {
+      const caseContent = await retrieveCases(elasticsearchClient, logger, index, {
+        id,
+        size,
+        owner,
+        priority,
+        closed,
+        caseNumber,
+        createdAfter,
+        createdBefore,
+        semanticQuery,
+        status,
+        updatedAfter,
+        updatedBefore,
+      });
+
+      logger.info(`Retrieved ${caseContent.length} support cases`);
 
-    const contentFragments = result.hits.hits.map((hit) => {
       return {
-        type: 'text' as const,
-        text:
-          hit.highlight?.infer_field
-            .flat()
-            .map((highlight) => highlight)
-            .join('\n') || '',
+        content: caseContent,
       };
+    }
+  );
+
+  return server;
+}
+
+/**
+ * Retrieves possible values for enum fields from Elasticsearch
+ */
+async function getFieldValues(
+  elasticsearchClient: ElasticsearchClient,
+  logger: Logger,
+  index: string,
+  enumFields: Field[]
+): Promise<FieldWithValues[]> {
+  let fieldValues: FieldWithValues[] = enumFields.map((field) => ({
+    ...field,
+    values: [],
+  }));
+
+  try {
+    const aggs = Object.fromEntries(
+      enumFields.map((field) => [
+        field.field,
+        {
+          terms: {
```

Comment:
suggestion: use https://www.elastic.co/guide/en/elasticsearch/reference/current/search-terms-enum.html as count doesnt matter.

Reply:
perfect, that's neat